tools/src/tokenize.rs

```diff
@@ -266,7 +266,77 @@ pub fn tokenize_c_like(string: &String, spec: &LanguageSpec) -> Vec<Token> {
                 tokens.push(Token {start: start, end: peek_pos(), kind: TokenKind::StringLiteral});
             }
             next_token_maybe_regexp_literal = true;
-        } else if ch == '\'' || ch == '"' {
+        } else if ch == 'R' {
+            // Handle raw literals.
+            assert!(peek_char() == '"');
+            get_char();
+
+            // Read delimiter.
+            loop {
+                let c = peek_char();
+                if c == '(' {
+                    break;
+                }
+
+                if c == ')' || c == '\\' || is_whitespace(c) {
+                    writeln!(&mut std::io::stderr(), "Illegal character in raw string delimiter").unwrap();
```

Comment:
Let's assume that the input is valid C++. We're already running clang on it before this step. Same for some of the other cases below.